core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java

```diff
@@ -1138,8 +1138,13 @@ protected final MaterializedViewFixture sql(String materialize,
 
   /** Test case for
    * <a href="https://issues.apache.org/jira/browse/CALCITE-6501">[CALCITE-6501]
-   * Assertion Error in JoinUnifyRule Due to Type Mismatch</a>. */
-  @Test void testLeftProjectOnRightJoinToJoinFailed() {
+   * Assertion Error in JoinUnifyRule Due to Type Mismatch</a>.
+   *
+   * <p>Originally, this test was noMat due to a type mismatch.
```

Comment:
I don't think this kind of comment is useful long term. It is useful for the reviewers of this PR, but not for maintaining the code. This comment would be better as a github comment rather than a code comment. Same for the subsequent comments.